admin/worker/deployments_health_check.go

```diff
@@ -38,54 +43,97 @@ func (w *Worker) deploymentsHealthCheck(ctx context.Context) error {
 				}
 				continue
 			}
+			addExpectedInstance(expectedInstances, d)
 			if seenHosts[d.RuntimeHost] {
 				continue
 			}
 			seenHosts[d.RuntimeHost] = true
 			group.Go(func() error {
-				return w.deploymentHealthCheck(cctx, d)
+				instances, ok := w.deploymentHealthCheck(cctx, d)
+				if ok {
+					mu.Lock()
+					actualInstances[d.RuntimeHost] = instances
+					mu.Unlock()
+				}
+				return nil
 			})
 		}
 		if err := group.Wait(); err != nil {
 			return err
 		}
 		if len(deployments) < limit {
-			return nil
+			break
 		}
 		afterID = deployments[len(deployments)-1].ID
 		// fetch again
 	}
+
+	// compare expected and actual instances
+	for host, expected := range expectedInstances {
+		actual, ok := actualInstances[host]
+		if !ok {
+			// runtime health check failed
+			continue
+		}
```

Comment:
Shouldn't it error here?

Reply:
No logging of errors is handled `deploymentHealthCheck` itself. 
Here it is just comparing expected and actual instances across all runtimes.

Reply:
Okay, I see